test-framework/core/src/main/java/org/keycloak/test/framework/realm/ClientSupplier.java

```diff
@@ -34,11 +36,22 @@ public ManagedClient getValue(InstanceContext<ManagedClient, InjectClient> insta
             clientRepresentation.setClientId(clientId);
         }
 
-        Response response = realm.admin().clients().create(clientRepresentation);
-        String uuid = ApiUtil.handleCreatedResponse(response);
-        clientRepresentation.setId(uuid);
+        List<ClientRepresentation> clients = realm.admin().clients().findByClientId(clientRepresentation.getClientId());
```

Comment:
Don't do this unless the client is not being created; otherwise we have an extra and wasteful request searching clients for all managed clients.

If the client is being created and there is a duplicate this will be thrown anyways as the create client will throw a 409.

Reply:
I see you are suggesting to then handle the status code and have a meaningful message as I'm proposing here.

My goal here is to provide a better message to developers otherwise you get a not so clear/friendly error and stacktrace.

Reply:
Fixed